rs/execution_environment/src/execution/replicated_query.rs

```diff
@@ -174,6 +185,30 @@ pub fn execute_replicated_query(
             .saturating_sub(output.num_instructions_left.get()),
     );
 
+    let (action, _) = canister
+        .system_state
+        .on_canister_result(
+            call_context_id,
+            None,
+            output.wasm_result.clone(),
+            instructions_used,
+        )
+        .unwrap();
```

Comment:
Why was this not needed before?

Reply:
This is used to close the call context that was opened earlier and it also helps us extract the refund that needs to be attached on the response (any leftover not accepted explicitly). Since we were not creating any call context before (it was a shortcut as we know that replicated queries cannot perform downstream calls, so we just wrapped the result of wasm execution in a response and we were done), this was not needed either.

Reply:
That makes sense: thank you for the explanation!